### ISSUE TITLE
Hide remote branches

### ISSUE THREAD

**Issue by rost314** [gitextensions/gitextensions#1198]
Removing the option to show remote branches from context menu just removes the labels from the view but not the commits reachable only by the remote branches.
It would be great if the commits would be filtered accordingly.

**Comment by johnddx** [comment]
Just wanted to note that this is still an issue.

**Comment by carlin-q-scott** [comment]
It's actually working how I want it to. I assumed like you did that it would hide the commits but it's only hiding the labels which is nice because now I can read commit messages where my local tracking branch matches the remote.

**Comment by johnddx** [comment]
Both things can be of value.  I want a way to actually hide remote branches so that I can more easily reason about all the branch tips I am working on locally.

**Comment by piotrbrzuska** [comment]
https://github.com/piotrbrzuska/gitextensions/commit/2bf22f70c8d96bbe1bc55bc68412414e373882c7

I do it by pass all local branches into branch filter. It works, but I don't now how this will work with a large number of branches.

**Comment by vbjay** [comment]
Look at the view menu. You can show only active branch. You can also
inactivate a remote.

**Comment by piotrbrzuska** [comment]
@vbjay yes and no, there is a option in view menu to show remote branches but it only hide labels for remote branches.
Now I testing my solutions, i found some errors, so I will try to fix it.

**Comment by vbjay** [comment]
I did say you could inactivate a remote.

**Comment by piotrbrzuska** [comment]
👍 its also very usefull!

**Comment by johndog** [comment]
It occurs to me that I don't understand this note from @carlin-q-scott:

> It's actually working how I want it to. I assumed like you did that it would hide the commits but it's only hiding the labels which is nice because now I can read commit messages where my local tracking branch matches the remote.

If you hide the remotes, how do you know which local tracking branches actually match the remote?  Do you quickly toggle with ctrl+shift+R?

I think having all the labels push the commit text off the screen is a problem in its own right.  Maybe there should be an abbreviated label mode, where you only see the _full_ branch name in a tool tip, and where remote and tracking branches are collapsed into one label if they are both on the same commit.

**Comment by johndog** [comment]
@piotrbrzuska: I have worked around this by manually typing "refs/heads/*" into the branch filter combo.  I wonder if that means you don't need to explicitly add _all_ branches to the filter, but you can just use an existing filter mechanism with a wildcard.

**Comment by carlin-q-scott** [comment]
@johndog 
> If you hide the remotes, how do you know which local tracking branches actually match the remote?

You should be able to tell which locals are behind their remote by looking at the commits above the branch label. If this feature worked as proposed by @rost314 then you wouldn't be able to do that and would have to resort to toggling the feature off and on.

>I think having all the labels push the commit text off the screen is a problem in its own right. Maybe there should be an abbreviated label mode, where you only see the full branch name in a tool tip, and where remote and tracking branches are collapsed into one label if they are both on the same commit.

Fair point and not a bad idea for a solution; personally I'd be in favor of collapsing synced branch labels and somehow indicating that happened with bold text/background color or something. I'd however want that feature implemented before updating the "hide remote branches" so that people dealing with long labels aren't left without a workaround to the long commit message problem.

**Comment by johndog** [comment]
Thanks, @carlin-q-scott 

> You should be able to tell which locals are behind their remote by looking at the commits above the branch label

Ah yes, didn't think of that.

I think the following:

- A fix like @piotrbrzuska's proposed fix, 
- Plus, renaming "Show remote branches" to "Show remote branch labels" 

would fully address the concern raised by @rost314, while preserving the current behavior.  

"Show remote branches" has a misleading name.  Up until that point in the menu the phrase "show branches" always affects which commits are displayed.  The app should adopt language that adequately distinguishes between commit graph filtering and filtering the labels that show.
